pkg/query/remote_engine.go

```diff
@@ -246,10 +246,10 @@ func (r *remoteQuery) Exec(ctx context.Context) *promql.Result {
 				F: s.Value,
 			})
 		}
-		for _, h := range ts.Histograms {
+		for _, hp := range ts.Histograms {
 			series.Histograms = append(series.Histograms, promql.HPoint{
-				T: h.Timestamp,
-				H: prompb.HistogramProtoToFloatHistogram(h),
+				T: hp.Timestamp,
+				H: prompb.FloatHistogramProtoToFloatHistogram(hp),
```

Comment:
Shouldn't we check `hp.IsFloatHistogram` before calling this?

Reply:
Queries should only return float histograms: https://github.com/thanos-io/thanos/pull/6323#discussion_r1202538124